LocalTransactionInterceptor does not match an interceptor bean (Payara Micro 4.1.1.161.1)

Deploying application on Payara Micro 4.1.1.161.1 gives the following error:
```
[2016-04-06T15:01:38.510+0200] [Payara 4.1] [SEVERE] [NCLS-CORE-00026] [javax.enterprise.system.core] [tid: _ThreadID=1 _ThreadName=main] [timeMillis: 1459947698510] [levelValue: 1000] [[
  Exception during lifecycle processing
org.glassfish.deployment.common.DeploymentException: CDI deployment failure:WELD-001417: Enabled interceptor class <class>org.mybatis.cdi.LocalTransactionInterceptor</class> in file:/tmp/gfembed3604631096618476727tmp/applications/my-mybatis-app/WEB-INF/classes/META-INF/beans.xml@7 does not match an interceptor bean: the class is not found, or not annotated with @Interceptor and still not registered through a portable extension, or not annotated with @Dependent inside an implicit bean archive
	at org.glassfish.weld.WeldDeployer.event(WeldDeployer.java:234)
	at org.glassfish.kernel.event.EventsImpl.send(EventsImpl.java:131)
	at org.glassfish.internal.data.ApplicationInfo.load(ApplicationInfo.java:329)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:496)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:219)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:487)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:539)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:535)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:534)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:565)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:557)
	at java.security.AccessController.doPrivileged(Native Method)
	at javax.security.auth.Subject.doAs(Subject.java:360)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:556)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1464)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:109)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1846)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1722)
	at com.sun.enterprise.admin.cli.embeddable.DeployerImpl.deploy(DeployerImpl.java:134)
	at fish.payara.micro.PayaraMicro.deployAll(PayaraMicro.java:931)
	at fish.payara.micro.PayaraMicro.bootStrap(PayaraMicro.java:717)
	at fish.payara.micro.PayaraMicro.main(PayaraMicro.java:105)
Caused by: org.jboss.weld.exceptions.DeploymentException: WELD-001417: Enabled interceptor class <class>org.mybatis.cdi.LocalTransactionInterceptor</class> in file:/tmp/gfembed3604631096618476727tmp/applications/my-mybatis-app/WEB-INF/classes/META-INF/beans.xml@7 does not match an interceptor bean: the class is not found, or not annotated with @Interceptor and still not registered through a portable extension, or not annotated with @Dependent inside an implicit bean archive
	at org.jboss.weld.bootstrap.Validator.validateEnabledInterceptorClasses(Validator.java:658)
	at org.jboss.weld.bootstrap.Validator.validateDeployment(Validator.java:481)
	at org.jboss.weld.bootstrap.WeldStartup.validateBeans(WeldStartup.java:443)
	at org.jboss.weld.bootstrap.WeldBootstrap.validateBeans(WeldBootstrap.java:90)
	at org.glassfish.weld.WeldDeployer.event(WeldDeployer.java:225)
	... 23 more
]]
```

It looks exactly like a problem described [here](https://java.net/jira/browse/GLASSFISH-20667). Changing `bean-discovery-mode` in mybatis-cdi `beans.xml` file to `bean-discovery-mode="all"` solves the problem.

--- a/src/main/resources/META-INF/beans.xml
+++ b/src/main/resources/META-INF/beans.xml
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <beans xmlns="http://xmlns.jcp.org/xml/ns/javaee" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/beans_1_1.xsd"
-    version="1.1" bean-discovery-mode="annotated">
+    version="1.1" bean-discovery-mode="all">
 </beans>

